### DISCUSSION TITLE
How to make the device connect to the wi-fi router without the app?

### DISCUSSION THREAD

**Discussion by ghost** [rytilahti/python-miio#1048] [Q&A] [answered]
How can I tell my air purifier to connect to my wifi? I know that there is something like configure_wifi, but I don't know how to use it.

**Comment by rytilahti** [comment, marked as answer]
Unfortunately there is no (easy, w/o having some secondary access to the device) way to do that due to how the onboarding process works. Everytime `configure_wifi` is issued, the device will generate a new token for communication that should be used for future communications but there is no way to access that using the API.

Dustcloud (https://github.com/dgiese/dustcloud) could be in theory used to capture the new token assuming that the new token gets send out to the cloud for communications, but I haven't heard any reports if that is really the case.
